data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+                    # print(f"Original frame count: {original_frame_count}")
+                    # print(f"Current frame count: {current_frame_count}")
+                    print(f"Updated pose shape before interpolation: {types_gloss_pose.body.data.shape}")
+
+                    # Interpolate pose if fps of dgs types pose is different from dgs corpus pose
+                    if types_gloss_pose.body.fps != original_pose_body.fps:
+                        interpolated_pose_body = types_gloss_pose.body.interpolate(new_fps=original_pose_body.fps)
+                        updated_pose_header = self._get_pose_header("dgs_types")
+                        updated_pose = Pose(updated_pose_header, interpolated_pose_body)
+                        print(f"Updated pose shape after interpolation: {interpolated_pose_body.data.shape}")
+                    else:
+                        updated_pose = types_gloss_pose
+            else:
+                updated_pose = original_pose
+
+            updated_poses_sequence.append(updated_pose)
+
+
+        # Concatenate poses
+        concatenated_original_pose = concatenate_poses(original_poses_sequence)
+        concatenated_updated_pose = concatenate_poses(updated_poses_sequence)
+
+        # print(f"Concatenated original pose shape: {concatenated_original_pose.body.data.shape}")
+        # print(f"Concatenated updated pose shape: {concatenated_updated_pose.body.data.shape}")
+
+        return concatenated_original_pose, concatenated_updated_pose, metadata, valid_gloss_count, replaced_gloss_count
+
+    def generate_dataset(self):
+        """
+        Processes the entire DGS Corpus and updates the pose sequences.
+        Returns the processed dataset.
+        """
+        print("Processing DGS Corpus...")
+
+        # Initialize counters and dataset list
+        self.dataset = []
+        self.total_signs = 0
+        self.replaced_signs = 0
+
+        for idx, data_entry in enumerate(self.corpus['train']):
+            # Limit number of examples to process
+            if self.num_examples and idx >= self.num_examples:
+                break
+
+            # Process each sentence entry
+            processed_data = self._process_sentence(data_entry)
+            self.dataset.append(processed_data)
+
+            # Update counters
+            self.total_signs += processed_data[3]
+            self.replaced_signs += processed_data[4]
+
+            # Log progress
+            print(f"Processed {idx + 1}/{self.num_examples or 'all'} sentences.")
+            print(f"Total signs: {self.total_signs}")
+            print(f"Replaced signs: {self.replaced_signs}")
+
+
+        # Final summary
+        print("Processing complete.")
+        print(f"Total sentences processed: {len(self.dataset)}")
+        print(f"Total signs: {self.total_signs}")
+        print(f"Replaced signs: {self.replaced_signs}")
+
+        return self.dataset
+
+
+
+
+# # Initialize the DGSDataset class
```

Comment:
i assume this is in a comment since you don't want it to run.
you should probably encapsulate it with `if __name__ == "__main__"`
ChatGPT could help with improving this for runtime / I use [Cursor](https://www.cursor.com/) IDE which gives a lot of help inline